dGame/dUtilities/Preconditions.cpp

```diff
@@ -137,7 +137,7 @@ bool Precondition::CheckValue(Entity* player, const uint32_t value, bool evaluat
 
 		return inventoryComponent->GetLotCount(value) >= count;
 	case PreconditionType::DoesNotHaveItem:
-		return inventoryComponent->IsEquipped(value) < count;
+		return inventoryComponent->IsEquipped(value) < static_cast<bool>(count);
```

Comment:
The comparison being performed here is really unclear to me. Is it equivalent to this?
```suggestion
		return inventoryComponent->IsEquipped(value) && count > 0;
```
If so, maybe it should be rewritten like that to make the intent clearer.

Reply:
sure